plugins/ipam/host-local/main.go

```diff
@@ -30,12 +31,66 @@ import (
 
 func main() {
 	// TODO: implement plugin version
-	skel.PluginMain(cmdAdd, cmdGet, cmdDel, version.All, "TODO")
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, "TODO")
 }
 
-func cmdGet(args *skel.CmdArgs) error {
-	// TODO: implement
-	return fmt.Errorf("not implemented")
+func loadNetConf(bytes []byte) (*types.NetConf, string, error) {
+	n := &types.NetConf{}
+	if err := json.Unmarshal(bytes, n); err != nil {
+		return nil, "", fmt.Errorf("failed to load netconf: %v", err)
+	}
+	return n, n.CNIVersion, nil
+}
+
+func cmdCheck(args *skel.CmdArgs) error {
+
+	ipamConf, _, err := allocator.LoadIPAMConfig(args.StdinData, args.Args)
+	if err != nil {
+		return err
+	}
+
+	// Get PrevResult from stdin... store in RawPrevResult
+	n, _, err := loadNetConf(args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	// Parse previous result.
+	if n.RawPrevResult == nil {
+		return fmt.Errorf("Required prevResult missing")
+	}
+
+	if err := version.ParsePrevResult(n); err != nil {
+		return err
+	}
+
+	result, err := current.NewResultFromResult(n.PrevResult)
+	if err != nil {
+		return err
+	}
+
+	// Each address in results.IPs must match what is in the container
+	for _, ips := range result.IPs {
```

Comment:
@mccv1r0 So we decided the following about the IPAM plugins in the last maintainer's meeting. The Spec says:

```
The plugin should return an error if a resource included in the CNI Result type (interface, address or route):
   - was created by the plugin, and
   - is listed in prevResult, and
   - does not exist, or is in an invalid state.
```

So for host-local, that would mean for each file we find in the network name's data directory that contains the ContainerID + interface name ("was created by the plugin") we look for that IP in prevResult ("is listed in prevResult"). If that IP is in prevResult, only then do we "validate" it.

But host-local intentionally doesn't handle anything about the interface, so it shouldn't check for the existence of the IP. So that means that host-local really doesn't have to CHECK anything about the IP addresses themselves and should probably just return "success".

*However*, it certainly should return an error if one of the two following conditions exists:

1) the data directory is not accessible (because the plugin was asked to allocate an IP on ADD there clearly must be a data directory or the ADD would have failed; so there must still be a data dir)
2) the ContainerID cannot be found in the data directory (the plugin would have allocated an IP on ADD so there clearly must be >= 1 file for the container in the data dir)

Does that all make sense?